### PR TITLE
Rename BaggageKeyPredicate to Filter 

### PR DESCRIPTION
**Pull request by MrAlias**
`baggagetrace.BaggageKeyPredicate` is overly verbose. Given this type is used as a filter, rename it to match its functionality.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -45,6 +45,7 @@ This project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.htm
 - Improve performance of `go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc` with the usage of `WithAttributeSet()` instead of `WithAttribute()`. (#5664)
 - Improve performance of `go.opentelemetry.io/contrib/instrumentation/net/http/otelhttp` with the usage of `WithAttributeSet()` instead of `WithAttribute()`. (#5664)
 - Update `go.opentelemetry.io/contrib/config` to latest released configuration schema which introduces breaking changes where `Attributes` is now a `map[string]interface{}`. (#5758)
+- Rename `BaggageKeyPredicate` to `Filter` in `go.opentelemetry.io/contrib/processors/baggage/baggagetrace`. (#TBD)
```
Comment (MrAlias) [inline CHANGELOG.md:48]:
```suggestion
- Rename `BaggageKeyPredicate` to `Filter` in `go.opentelemetry.io/contrib/processors/baggage/baggagetrace`. (#5809 )
```

File: processors/baggage/baggagetrace/processor.go [hunk 2]
```diff
@@ -11,15 +11,16 @@ import (
...
 
-// BaggageKeyPredicate is a function that returns true if the baggage key should be added to the span.
-type BaggageKeyPredicate func(baggageKey string) bool
+// Filter returns true if the baggage member with key should be added to a
+// span.
+type Filter func(key string) bool
 
-// AllowAllBaggageKeys is a BaggageKeyPredicate that allows all baggage keys.
-var AllowAllBaggageKeys = func(string) bool { return true }
+// AllowAllBaggageKeys allows all baggage members to be added to a span.
```
Comment (pellared) [inline processors/baggage/baggagetrace/processor.go:18]: Is there any reason why the filter doesn't accept a baggage.Member instead of a key?

Comment (MrAlias) [reply processors/baggage/baggagetrace/processor.go:18]: I don't see why we couldn't do that in general. Though, I would like to keep this PR scoped to just the rename.

We can address that in a follow up PR though.

Comment (MrAlias) [reply processors/baggage/baggagetrace/processor.go:18]: https://github.com/open-telemetry/opentelemetry-go-contrib/pull/5813

**Review by pellared — APPROVED** [review]

**Review by dmathieu — APPROVED** [review]

**Pull request merged**